superset-frontend/src/explore/components/DisplayQueryButton/DisplayQueryButton.test.tsx

```diff
@@ -0,0 +1,230 @@
+    form_data: {
+      adhoc_filters: [],
+      all_columns_x: ['age'],
+      color_scheme: 'supersetColors',
+      datasource: '49__table',
+      granularity_sqla: 'time_start',
+      groupby: null,
+      label_colors: {},
+      link_length: '25',
+      queryFields: { groupby: 'groupby' },
+      row_limit: 10000,
+      slice_id: 318,
+      time_range: 'No filter',
+      time_range_endpoints: ['inclusive', 'exclusive'],
+      url_params: {},
+      viz_type: 'histogram',
+      x_axis_label: 'age',
+      y_axis_label: 'count',
+    },
+    modified: '<span class="no-wrap">3 days ago</span>',
+    owners: [],
+    slice_id: 318,
+    slice_name: 'Age distribution of respondents',
+    slice_url: '/superset/explore/?form_data=%7B%22slice_id%22%3A%20318%7D',
+  },
+  chartStatus: 'rendered',
+  onOpenPropertiesModal: jest.fn(),
+  onOpenInEditor: jest.fn(),
+};
+
+beforeEach(() => {
+  props = {
+    latestQueryFormData: {
+      viz_type: 'histogram',
+      datasource: '49__table',
+      slice_id: 318,
+      url_params: {},
+      time_range_endpoints: ['inclusive', 'exclusive'],
+      granularity_sqla: 'time_start',
+      time_range: 'No filter',
+      all_columns_x: ['age'],
+      adhoc_filters: [],
+      row_limit: 10000,
+      groupby: null,
+      color_scheme: 'supersetColors',
+      label_colors: {},
+      link_length: '25',
+      x_axis_label: 'age',
+      y_axis_label: 'count',
+    },
+    slice: {
+      cache_timeout: null,
+      changed_on: '2021-03-19T16:30:56.750230',
+      changed_on_humanized: '3 days ago',
+      datasource: 'FCC 2018 Survey',
+      description: null,
+      description_markeddown: '',
+      edit_url: '/chart/edit/318',
+      form_data: {
+        adhoc_filters: [],
+        all_columns_x: ['age'],
+        color_scheme: 'supersetColors',
+        datasource: '49__table',
+        granularity_sqla: 'time_start',
+        groupby: null,
+        label_colors: {},
+        link_length: '25',
+        queryFields: { groupby: 'groupby' },
+        row_limit: 10000,
+        slice_id: 318,
+        time_range: 'No filter',
+        time_range_endpoints: ['inclusive', 'exclusive'],
+        url_params: {},
+        viz_type: 'histogram',
+        x_axis_label: 'age',
+        y_axis_label: 'count',
+      },
+      modified: '<span class="no-wrap">3 days ago</span>',
+      owners: [],
+      slice_id: 318,
+      slice_name: 'Age distribution of respondents',
+      slice_url: '/superset/explore/?form_data=%7B%22slice_id%22%3A%20318%7D',
+    },
+    chartStatus: 'rendered',
+    onOpenPropertiesModal: jest.fn(),
+    onOpenInEditor: jest.fn(),
+  };
+});
+
+test('Shoud render a button', () => {
+  render(<DisplayQueryButton {...props} />, { useRedux: true });
+  expect(screen.getByRole('button')).toBeInTheDocument();
+});
+
+test('Shoud open a menu', () => {
+  render(<DisplayQueryButton {...props} />, {
+    useRedux: true,
+  });
+
+  expect(props.onOpenInEditor).toBeCalledTimes(0);
```

Comment:
I am not sure I am seeing the reason for testing whether these functions are called at this point?

Reply:
The idea is to ensure that if there is a change in the code, the behavior of when the actions are performed will be maintained. If by any accident a bug is introduced where and these actions are performed unintentionally, this test would capture that error.

The idea of this test is to ensure that only the functions that need to be performed are being performed.

Reply:
I believe for the specific case it might be a little bit far-fetched, but it's fine as it does not really compromise the quality of the tests. I'll approve once Michael's comments are handled. Thank you!